spec/models/target_spec.rb

```diff
@@ -32,74 +36,127 @@
     end
   end
 
-  describe '#get_progress' do
-    it 'returns zero if there are no registrations' do
-      conference = build(:conference)
-      target = build(:target, target_count: 10, unit: Target.units[:registrations])
-      conference.targets = [target]
-
-      expect(target.get_progress).to eq('0')
+  describe 'default scope' do
+    before do
+      @first_target = create(:target, due_date: 2.days.from_now)
+      @second_target = create(:target, due_date: 3.days.from_now)
     end
 
-    it 'returns 10 if there one registrations of 10' do
-      conference = create(:conference)
-      target = create(:target, target_count: 10, unit: Target.units[:registrations])
-      registration = create(:registration)
+    it 'orders by ascending due_date' do
+      expect(Target.all).to match_array [@first_target, @second_target]
+    end
+  end
 
-      conference.targets = [target]
-      conference.registrations = [registration]
+  describe 'associations' do
+    it { should belong_to(:conference) }
+    it { should belong_to(:campaign) }
+  end
 
-      expect(target.get_progress).to eq('10')
+  describe '#get_progress' do
+    it 'returns zero if there are no registrations' do
+      expect(registration_target.get_progress).to eq('0')
     end
 
-    it 'returns zero if there are no submissions' do
-      conference = create(:conference)
-      target = build(:target, target_count: 10, unit: Target.units[:submissions])
-      conference.targets = [target]
+    it 'returns 10 if there is one registration of 10' do
+      create(:registration, conference: registration_target.conference)
 
-      expect(target.get_progress).to eq('0')
+      expect(registration_target.get_progress).to eq('10')
     end
 
-    it 'returns 10 if there one submissions of 10' do
-      conference = create(:conference)
-      target = create(:target, target_count: 10, unit: Target.units[:submissions])
-      event = create(:event)
+    it 'returns zero if there are no submissions' do
+      expect(submission_target.get_progress).to eq('0')
+    end
 
-      conference.targets = [target]
-      conference.program.events = [event]
+    it 'returns 10 if there is one submission of 10' do
+      submission_target.conference.program.events = [event]
 
```

Comment:
I would create the event here. This line gets very confusing. 
You can just do `create(:event, program: conference.program)`, you can define the conference in the beginning and create targets of this particular conference. Same for line 81.